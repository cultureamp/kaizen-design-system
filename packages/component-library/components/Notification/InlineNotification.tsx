import * as React from "react"
import GenericNotification, {
  NotificationType,
} from "./components/GenericNotification"

type Props = {
  type: NotificationType
  title: string
  children: React.ReactNode
  autohide?: boolean
  autohideDelay?: "short" | "long"
  persistent: boolean
  hideCloseIcon: boolean
  onHide?: () => void
  automationId?: string
  noBottomMargin?: boolean
}

const InlineNotification = ({
  persistent,
  hideCloseIcon,
  ...otherProps
}: Props) => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    {...otherProps}
  />
)

InlineNotification.defaultProps = {
  autohideDelay: "short",
  autohide: false,
  hideCloseIcon: false,
  persistent: false,
}

export default InlineNotification
