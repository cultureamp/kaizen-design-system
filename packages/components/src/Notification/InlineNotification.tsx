import React, { HTMLAttributes } from "react"
import classnames from "classnames"
import { OverrideClassName } from "~types/OverrideClassName"
import styles from "./InlineNotification.module.scss"

/**
 * @todo: Refer to `docs/` for further code standards and guidelines
 */

/**
 * @todo: Replace `HTMLAttributes<HTMLDivElement>` with attributes/props you need to extend
 */
export type InlineNotificationProps = {
  /** @todo: Add custom props here */
  exampleRequiredString: string
  isReversed?: boolean
} & OverrideClassName<HTMLAttributes<HTMLDivElement>>

export const InlineNotification = ({
  exampleRequiredString,
  isReversed = false,
  classNameOverride,
  ...restProps
}: InlineNotificationProps): JSX.Element => {
  return (
    <div
      className={classnames(
        styles.inlineNotification,
        isReversed && styles.isReversed,
        classNameOverride,
      )}
      {...restProps}
    >
      {exampleRequiredString}
    </div>
  )
}

InlineNotification.displayName = "InlineNotification"
