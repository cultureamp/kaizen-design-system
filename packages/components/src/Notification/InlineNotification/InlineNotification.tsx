import React from "react"
import classnames from "classnames"
import { HeadingProps } from "~components/Heading"
import { OverrideClassName } from "~types/OverrideClassName"
import {
  GenericNotification,
  GenericNotificationType,
  GenericNotificationVariant,
} from "../subcomponents/GenericNotification"
import styles from "../subcomponents/GenericNotification/GenericNotification.module.scss"

export type InlineNotificationBase = OverrideClassName<{
  children?: React.ReactNode
  /** @default false */
  persistent?: boolean
  /** @default false */
  hideCloseIcon?: boolean
  onHide?: () => void
  noBottomMargin?: boolean
  forceMultiline?: boolean
  headingProps?: HeadingProps
  isSubtle?: boolean
}>

export type InlineNotificationProps = InlineNotificationBase &
  (GenericNotificationType | GenericNotificationVariant)

/**
 * {@link https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082093392/Inline+Notification Guidance} |
 * {@link https://cultureamp.design/storybook/?path=/docs/components-notifications-inline-notification--docs Storybook}
 */
export const InlineNotification = ({
  isSubtle,
  hideCloseIcon = false,
  persistent = false,
  classNameOverride,
  ...otherProps
}: InlineNotificationProps): JSX.Element => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    classNameOverride={classnames(classNameOverride, [
      isSubtle && styles.subtle,
    ])}
    {...otherProps}
  />
)

InlineNotification.displayName = "InlineNotification"
