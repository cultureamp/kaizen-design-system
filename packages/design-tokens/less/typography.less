/** THIS IS AN AUTOGENERATED FILE **/

@kz-typography-display-0-font-family: "Greycliff CF";
@kz-typography-display-0-font-weight: 700;
@kz-typography-display-0-font-size: 4.5rem;
@kz-typography-display-0-line-height: 4.5rem;
@kz-typography-display-0-letter-spacing: -0.03em;
@kz-typography-heading-1-font-family: "Greycliff CF";
@kz-typography-heading-1-font-weight: 700;
@kz-typography-heading-1-font-size: 2.25rem;
@kz-typography-heading-1-line-height: 2.25rem;
@kz-typography-heading-1-letter-spacing: -0.03em;
@kz-typography-heading-2-font-family: "Greycliff CF";
@kz-typography-heading-2-font-weight: 700;
@kz-typography-heading-2-font-size: 1.875rem;
@kz-typography-heading-2-line-height: 2.25rem;
@kz-typography-heading-2-letter-spacing: -0.03em;
@kz-typography-heading-3-font-family: "Greycliff CF";
@kz-typography-heading-3-font-weight: 700;
@kz-typography-heading-3-font-size: 1.375rem;
@kz-typography-heading-3-line-height: 1.5rem;
@kz-typography-heading-3-letter-spacing: -0.03em;
@kz-typography-heading-4-font-family: "Inter";
@kz-typography-heading-4-font-weight: 600;
@kz-typography-heading-4-font-size: 1.125rem;
@kz-typography-heading-4-line-height: 1.5rem;
@kz-typography-heading-4-letter-spacing: normal;
@kz-typography-heading-5-font-family: "Inter";
@kz-typography-heading-5-font-weight: 600;
@kz-typography-heading-5-font-size: 1rem;
@kz-typography-heading-5-line-height: 1.5rem;
@kz-typography-heading-5-letter-spacing: normal;
@kz-typography-heading-6-font-family: "Inter";
@kz-typography-heading-6-font-weight: 700;
@kz-typography-heading-6-font-size: 0.875rem;
@kz-typography-heading-6-line-height: 1.5rem;
@kz-typography-heading-6-letter-spacing: normal;
@kz-typography-paragraph-intro-lede-font-family: "Inter";
@kz-typography-paragraph-intro-lede-font-weight: 400;
@kz-typography-paragraph-intro-lede-font-size: 1.25rem;
@kz-typography-paragraph-intro-lede-line-height: 1.875rem;
@kz-typography-paragraph-intro-lede-letter-spacing: -0.5px;
@kz-typography-paragraph-body-font-family: "Inter";
@kz-typography-paragraph-body-font-weight: 400;
@kz-typography-paragraph-body-font-size: 1rem;
@kz-typography-paragraph-body-line-height: 1.5rem;
@kz-typography-paragraph-body-letter-spacing: normal;
@kz-typography-paragraph-small-font-family: "Inter";
@kz-typography-paragraph-small-font-weight: 400;
@kz-typography-paragraph-small-font-size: 0.875rem;
@kz-typography-paragraph-small-line-height: 1.125rem;
@kz-typography-paragraph-small-letter-spacing: normal;
@kz-typography-paragraph-extra-small-font-family: "Inter";
@kz-typography-paragraph-extra-small-font-weight: 400;
@kz-typography-paragraph-extra-small-font-size: 0.75rem;
@kz-typography-paragraph-extra-small-line-height: 1.125rem;
@kz-typography-paragraph-extra-small-letter-spacing: normal;
