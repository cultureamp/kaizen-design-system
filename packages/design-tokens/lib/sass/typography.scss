/** THIS IS AN AUTOGENERATED FILE **/

$font-heading-family: 'Greycliff CF';
$font-heading-size--0: 4.5rem;
$font-heading-size--1: 2.25rem;
$font-heading-size--2: 1.875rem;
$font-heading-size--3: 1.375rem;
$font-heading-size--4: 1.125rem;
$font-heading-size--5: 1rem;
$font-heading-size--6: 0.875rem;
$font-paragraph-family: 'Inter';
$font-paragraph-size--4: 1.25rem;
$font-paragraph-size--5: 1rem;
$font-paragraph-size--6: 0.875rem;
$font-paragraph-size--7: 0.75rem;
$font-preformatted-family: 'IBM Plex Mono';
$font-preformatted-size--5: 1;
$DEPRECATED-font-default-family: 'Open Sans';
$DEPRECATED-font-sans-serif-family: 'Ideal Sans A';
