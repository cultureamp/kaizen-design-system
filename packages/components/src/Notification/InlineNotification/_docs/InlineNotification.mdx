import { Canvas, Controls, Meta, Description} from "@storybook/blocks"
import { KAIOInstallation, ResourceLinks } from "~storybook/components"
import * as InlineNotificationExamples from "./InlineNotification.stories"

<Meta title="Components/Notifications/Inline Notification" />

# Inline Notification

<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/Notification/InlineNotification"
  figma="https://www.figma.com/file/eZKEE5kXbEMY3lx84oz8iN/%F0%9F%92%9C-UI-Kit%3A-Heart?type=design&node-id=1929%3A21594&mode=design&t=HmSOpdHdka19ZmTO-1"
  designGuidelines="https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082093392/Inline+Notification"

/>

<KAIOInstallation exportNames="InlineNotification" />

## Overview

`<InlineNotification>` is used to display contextual information, including positive feedback or confirmation about an action, warnings about possible issues or show-stopping errors.

<Canvas of={InlineNotificationExamples.Playground} />

## API

<Controls of={InlineNotificationExamples.Playground} />


### persistent

<Description of={InlineNotificationExamples.Persistent} />
<Canvas of={InlineNotificationExamples.Persistent} />

### forceMultiline

<Description of={InlineNotificationExamples.ForceMultilineDemo} />
<Canvas of={InlineNotificationExamples.ForceMultilineDemo} />

<Description of={InlineNotificationExamples.ContentMultilineDemo} />
<Canvas of={InlineNotificationExamples.ContentMultilineDemo} />
