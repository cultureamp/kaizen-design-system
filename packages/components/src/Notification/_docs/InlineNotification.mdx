import { Canvas, Controls, Description, Meta } from "@storybook/blocks"
import { ResourceLinks, KaioNotification, Installation } from "~storybook/components"
import * as InlineNotificationStories from "./InlineNotification.stories"

<Meta of={InlineNotificationStories} />

# InlineNotification

{/*
  * @todo:
  * - Adjust Github link (if needed)
  * - Add Figma and Design Guidelines (Confluence) link
  */}
<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/InlineNotification"
  figma=""
  designGuidelines=""
  className="!mb-8"
/>

<KaioNotification />

<Installation
  installCommand="yarn add @kaizen/components"
  importStatement='import { InlineNotification } from "@kaizen/components"'
/>

## Overview

{/* @todo: Update summary. */}
Brief summary of the component here.

<Canvas of={InlineNotificationStories.Playground} />
<Controls of={InlineNotificationStories.Playground} />

## API

{/*
  * @todo: Describe props and provide example stories.
  * You can choose to write in MDX or to use <Description />.
  */}
### Reversed
<Description of={InlineNotificationStories.Reversed} />
<Canvas of={InlineNotificationStories.Reversed} />
