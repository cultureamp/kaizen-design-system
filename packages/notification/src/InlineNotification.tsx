import React from "react"
import { HeadingProps } from "@kaizen/typography"
import GenericNotification, {
  NotificationType,
} from "./components/GenericNotification"
import styles from "./components/GenericNotification.module.scss"

export type InlineNotificationProps = {
  type: NotificationType
  children?: React.ReactNode
  autohide?: boolean
  autohideDelay?: "short" | "long"
  persistent: boolean
  hideCloseIcon: boolean
  onHide?: () => void
  automationId?: string
  noBottomMargin?: boolean
  forceMultiline?: boolean
  headingProps?: HeadingProps
  isSubtle?: boolean
  /**
   * **Deprecated:** Use headingProps
   * @deprecated
   */
  title?: string
}

/**
 * {@link https://cultureamp.design/components/inline-notification/ Guidance} |
 * {@link https://cultureamp.design/storybook/?path=/docs/components-notification-inline-notification--default-kaizen-demo Storybook}
 */
export const InlineNotification = ({
  persistent,
  hideCloseIcon,
  isSubtle,
  ...otherProps
}: InlineNotificationProps): JSX.Element => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    classNameOverride={isSubtle ? styles.subtle : undefined}
    {...otherProps}
  />
)

InlineNotification.defaultProps = {
  autohideDelay: "short",
  autohide: false,
  hideCloseIcon: false,
  persistent: false,
}

InlineNotification.displayName = "InlineNotification"
