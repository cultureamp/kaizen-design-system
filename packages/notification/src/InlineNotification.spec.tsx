import React from "react"
import { render, screen } from "@testing-library/react"
import { HeadingProps } from "@kaizen/typography"
import { InlineNotification } from "./InlineNotification"

describe("<InlineNotification />", () => {
  it("renders a basic notification correctly", () => {
    const { container } = render(
      <InlineNotification type="cautionary" title="Warning">
        Something has gone wrong
      </InlineNotification>
    )

    expect(container.querySelector(".cancel")).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders a persistent notifications", () => {
    const { container } = render(
      <InlineNotification type="negative" title="Error" persistent>
        Something has gone wrong
      </InlineNotification>
    )

    expect(container.querySelector(".cancel")).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders a notification with a custom heading level", () => {
    const customHeadingProps: HeadingProps = {
      variant: "heading-6",
      tag: "h2",
      children: "Custom",
    }
    render(
      <InlineNotification
        type="positive"
        headingProps={customHeadingProps}
        persistent
      >
        Something has gone wrong
      </InlineNotification>
    )

    expect(screen.getByRole("heading", { level: 2 })).toBeInTheDocument()
  })
})
