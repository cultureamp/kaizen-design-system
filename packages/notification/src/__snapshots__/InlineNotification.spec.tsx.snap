// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineNotification /> renders a basic notification correctly 1`] = `
<div
  class="notification cautionary inline hidden"
  data-automation-class="generic-notification cautionary inline"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <div
      class="heading heading-6 notificationTitle dark small"
    >
      Warning
    </div>
    <div
      class="text"
    >
      Something has gone wrong
    </div>
  </div>
  <button
    class="cancel"
    data-testid="close-button"
    type="button"
  >
    <svg
      aria-label="close notification"
      class="icon"
      focusable="false"
      role="img"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </button>
</div>
`;

exports[`<InlineNotification /> renders a persistent notifications 1`] = `
<div
  class="notification negative inline hidden"
  data-automation-class="generic-notification negative inline"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <div
      class="heading heading-6 notificationTitle dark small"
    >
      Error
    </div>
    <div
      class="text"
    >
      Something has gone wrong
    </div>
  </div>
</div>
`;
